### PR TITLE
add support for gcr buckets with uniform_bucket_level_access = true

### PR DESCRIPTION
**Pull request by patricklubach**
As requested by @anouarchattouna I create another PR with the same changes from @andyroyle because @andyroyle does not respond anymore.

### CODE REVIEW

File: locals.tf [hunk 1]
```diff
@@ -67,4 +67,17 @@ locals {
...
+  buckets = [
+    for repo in var.gcr_repositories : repo.storage_region != null ? "${repo.storage_region}.artifacts.${repo.project_id != null ? repo.project_id : local.google_project_id}.appspot.com" : "artifacts.${repo.project_id != null ? repo.project_id : local.google_project_id}.appspot.com"
+  ]
+
+  # Buckets having uniform_bucket_level_access = false
+  google_storage_bucket_iam_member = [
+    for bucket in local.buckets : bucket if data.google_storage_bucket.bucket[bucket].uniform_bucket_level_access
+  ]
+
+  google_storage_bucket_access_control = [
```
Comment (anouarchattouna) [inline locals.tf:80]:
```suggestion
  # Buckets having uniform_bucket_level_access = false
  google_storage_bucket_access_control = [
```

File: locals.tf [hunk 2]
```diff
@@ -67,4 +67,17 @@ locals {
       }
     ]
   ])
+
+  buckets = [
+    for repo in var.gcr_repositories : repo.storage_region != null ? "${repo.storage_region}.artifacts.${repo.project_id != null ? repo.project_id : local.google_project_id}.appspot.com" : "artifacts.${repo.project_id != null ? repo.project_id : local.google_project_id}.appspot.com"
+  ]
+
+  # Buckets having uniform_bucket_level_access = false
```
Comment (anouarchattouna) [inline locals.tf:75]:
```suggestion
  # Buckets having uniform_bucket_level_access = true
```

**Review by anouarchattouna — CHANGES_REQUESTED** [review]
Can you also add this to `data.tf` please ?
```hcl
data "google_storage_bucket" "bucket" {
  for_each = toset(local.buckets)

  name = each.value
} 

```

File: iam.tf [hunk 3]
```diff
@@ -1,14 +1,23 @@
 # Grant cleaner service account access to delete references in Google Container Registry
+# for old-style ACL buckets
```
Comment (anouarchattouna) [inline iam.tf:2]:
```suggestion
# for buckets with uniform_bucket_level_access = false
```

File: data.tf [hunk 4]
```diff
@@ -1,2 +1,8 @@
 # get project details
 data "google_project" "this" {}
+
+data "google_storage_bucket" "bucket" {
+  for_each = toset(local.buckets)
+
+  name = each.value
+}
```
Comment (anouarchattouna) [inline data.tf:8]:
```suggestion
}

```

**Review by anouarchattouna — APPROVED** [review]
Last minor change, otherwise LGTM!

**Review by madhifallah — APPROVED** [review]
LGTM

**Pull request merged**